File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
const table = {};
  
  for (let i = 0, n = nums.length; i < n; i++) {
    const num = nums[i];
    const diff = target - num;
    
    if (table[diff] >= 0) {
      return [table[diff], i];
    } else {
      table[num] = i;
    }
  }
};

File: 1-two-sum/1-two-sum.rb
 def two_sum(nums, target)
      diff_lookup = {}
   
    nums.each_with_index do |num, i|
        diff = target - num
       if diff_lookup[diff].nil?
         diff_lookup[num] = i
       else
         return [diff_lookup[diff], i]
         # testing out codespace 
       end
     end
   end